ipn/ipnlocal/c2n.go

```diff
@@ -115,17 +115,21 @@ func (b *LocalBackend) handleC2NUpdate(w http.ResponseWriter, r *http.Request) {
 	// TODO(bradfitz): add some sort of semaphore that prevents two concurrent
 	// updates, or if one happened in the past 5 minutes, or something.
 
-	var res tailcfg.C2NUpdateResponse
-	res.Enabled = envknob.AllowsRemoteUpdate()
-	res.Supported = runtime.GOOS == "windows" || (runtime.GOOS == "linux" && distro.Get() == distro.Debian)
-
-	switch r.Method {
-	case "GET", "POST":
-	default:
+	// We interpret GET as a request for current status, and POST as a request to
+	// actually begin an update.
+	if r.Method != "GET" && r.Method != "POST" {
 		http.Error(w, "bad method", http.StatusMethodNotAllowed)
 		return
 	}
 
+	var res tailcfg.C2NUpdateResponse
+	res.Enabled = envknob.AllowsRemoteUpdate()
+
+	// We support `tailscale update` on most Linux distributions that we create
+	// packages for. Arch is a special case (see
+	// https://github.com/tailscale/tailscale/issues/6995).
+	res.Supported = runtime.GOOS == "windows" || (runtime.GOOS == "linux" && distro.Get() != distro.Arch)
```

Comment:
There's also Synology, macOS, FreeBSD.
I think we can invert this to only list unsupported platforms (Arch, iOS, Android?)
Or better - make a helper function in `clientupdate` (maybe identical to `func Update` but without executing the actual update)

Reply:
macOS macsys may require some fanciness to get working. I intend to update this for macOS when that's ready.

I'll do FreeBSD and Synology in this PR; coming up.

Reply:
We also have macOS app store version support here: https://github.com/tailscale/tailscale/blob/main/clientupdate/clientupdate.go#L556